src/main/java/org/folio/rest/impl/SamlAPI.java

```diff
@@ -216,22 +218,22 @@ private void doPostSamlCallback(String body, RoutingContext routingContext, Map<
 
     final SessionStore sessionStore = new DummySessionStore(routingContext.vertx(), routingContext.session());
     final VertxWebContext webContext = new VertxWebContext(routingContext, sessionStore);
-    final String relayState = getRelayState(routingContext, body);
-
+    final Cookie relayStateCookie = routingContext.request().getCookie(RELAY_STATE);
+    String relayStateCookieValue = "null";
     URI relayStateUrl;
     try {
-      assert (relayState != null); // this avoids a Sonar warning later on
-      relayStateUrl = new URI(relayState);
+      relayStateCookieValue = relayStateCookie.getValue();
+      relayStateUrl = new URI(relayStateCookieValue);
     } catch (Exception e) {
```

Comment:
I don't love that this is catching not just the two exceptions that are known to be thrown by `new URI` since it would mask other exceptions, but since it is only trying the two lines it doesn't make any real difference I suppose.